code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -38,25 +42,35 @@
 	move_delay = FALSE
 
 /obj/structure/closet/cardboard/open(mob/living/user, force = FALSE)
-	if(opened || !can_open(user, force))
-		return FALSE
-	var/list/alerted = null
-	if(egged < world.time)
-		var/mob/living/Snake = null
-		for(var/mob/living/L in src.contents)
-			Snake = L
-			break
-		if(Snake)
-			alerted = viewers(7,src)
-	..()
-	if(LAZYLEN(alerted))
-		egged = world.time + SNAKE_SPAM_TICKS
-		for(var/mob/living/L in alerted)
-			if(!L.stat)
-				if(!L.incapacitated(IGNORE_RESTRAINTS))
-					L.face_atom(src)
-				L.do_alert_animation()
-		playsound(loc, 'sound/machines/chime.ogg', 50, FALSE, -5)
+	var/do_alert = (COOLDOWN_FINISHED(src, alert_cooldown) && (locate(/mob/living) in contents))
+
+	if(!do_alert)
+		return ..()
+
+	// Cache the list before we open the box.
+	var/list/alerted = viewers(7, src)
+
+	// There are no mobs to alert?
+	if(!(locate(/mob/living) in alerted))
+		return ..()
+
+	. = ..()
+
+	// Box didn't open?
+	if(!.)
+		return
+
+	COOLDOWN_START(src, alert_cooldown, time_between_alerts)
+
+	for(var/mob/living/alerted_mob in alerted)
+		if(alerted_mob.stat == CONSCIOUS)
+			if(!alerted_mob.incapacitated(IGNORE_RESTRAINTS))
+				alerted_mob.face_atom(src)
+			alerted_mob.do_alert_animation()
+
+	playsound(loc, 'sound/machines/chime.ogg', 50, FALSE, -5)
+
+	return
```

Comment:
return at the end of the proc?

Reply:
Sorry, I have been doing stuff on more civilised code projects that don't have implicit returns for non void types.

I'll remove it shortly.